core/src/main/java/org/infinispan/cache/impl/SimpleCacheImpl.java

```diff
@@ -491,6 +491,25 @@ public int size() {
       return size;
    }
 
+   @Override
+   public CompletableFuture<Long> sizeAsync() {
+      // we have to iterate in order to provide precise result in case of expiration
+      long now = Long.MIN_VALUE;
+      long size = 0;
+      DataContainer<K, V> dataContainer = getDataContainer();
```

Comment:
I'm wondering, about this logic. If instead of calling it sizeAsync we called it "estimateSize", with an async method...
would we need to iterate this way to avoid counting expired entries?

Reply:
I am all for doing this and have wanted to for a while, however this method doesn't yet do this. We can add this for 10.1.